### PR TITLE
Added a command to display the versions for the Toolkit, VS Code, and OS

### PR DESCRIPTION
**Pull request by gsweetwood**
## Description

Created a command called getToolkitInfo.  This can be called with the command palette as "AWS: Get AWS Toolkit Info".  This creates a modal that displays the user's Toolkit version, VS Code version, and OS type and version.  The user has the option to copy that info to their clipboard.  This info is also displayed as output for the 'AWS Toolkit'.

## Motivation and Context

The template for issuing new tickets for the toolkit asks the user for this information.  This allows the user to gather that info in one command and can paste it in if they would like.  This will hopefully reduce the cost of writing up a new issue.

## Screenshot
Command palette for getToolkitInfo
Toolkit Info modal and output

## Testing

Passes all current tests. 

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.nls.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "AWS.command.deleteLambda.confirm": "Are you sure you want to delete lambda function '{0}'?",
     "AWS.command.deleteLambda.error": "There was an error deleting lambda function '{0}'",
     "AWS.command.deploySamApplication": "Deploy SAM Application",
+    "AWS.command.getToolkitInfo": "Get AWS Toolkit Info",
```
Comment (bryceitoc9) [inline package.nls.json]: I wonder if we should change this to something like "About the AWS Toolkit..."

File: package.nls.json [hunk 2]
```diff
@@ -143,6 +144,8 @@
     "AWS.message.prompt.quickStart.toastMessage": "You are now using the AWS Toolkit for Visual Studio Code, version {0}",
     "AWS.message.prompt.region.hide.title": "Select a region to hide from the AWS Explorer",
     "AWS.message.prompt.region.show.title": "Select a region to show in the AWS Explorer",
+    "AWS.message.prompt.copyButtonLabel": "Copy",
+    "AWS.message.toolkitInfo": "AWS.message.toolkitInfo','OS:  {0} {1} {2}\nVisual Studtio Code Version:  {3}\nAWS Toolkit for Visual Studio Code Version:  {4}\n",
```
Comment (bryceitoc9) [inline package.nls.json]: Assuming `AWS.message.toolkitInfo','` is not meant to be in this file?

File: .changes/next-release/Feature-546b77ba-20be-4573-811d-6a6d86b47f00.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+    "type": "Feature",
+    "description": "Created a command to display Toolkit and environment info. These include OS, VS Code, and Toolkit versions."
```
Comment (awschristou) [inline .changes/next-release/Feature-546b77ba-20be-4573-811d-6a6d86b47f00.json]: Consider something like the following to both inform and guide:
Added "COMMAND_NAME" Command to show AWS Toolkit versioning details that are useful to include with bug reports.

File: package.nls.json (hunk 2 above)
Comment (awschristou) [reply package.nls.json]: Typo: 'Studtio'

File: src/shared/extensionUtilities.ts [hunk 4]
```diff
@@ -202,3 +203,33 @@ export function toastNewUser(context: vscode.ExtensionContext): void {
...
+
+export async function getToolkitInfo(toolkitOutputChannel: vscode.OutputChannel) {
+    const copyButtonLabel = localize('AWS.message.prompt.copyButtonLabel', 'Copy')
+    const osType = os.type()
+    const osArch = os.arch()
+    const osRelease = os.release()
+    const vsCodeVersion = vscode.version
+    const envDetails = localize(
+        'AWS.message.toolkitInfo',
+        'OS:  {0} {1} {2}\nVisual Studtio Code Version:  {3}\nAWS Toolkit for Visual Studio Code Version:  {4}\n',
```
Comment (awschristou) [inline src/shared/extensionUtilities.ts]: typo 'Studtio '

**Comment by hunterwerlla** [comment]
Can you update the PR template to include this as a step

File: src/shared/extensionUtilities.ts (hunk 4 above)
Comment (hunterwerlla) [inline src/shared/extensionUtilities.ts]: typo: "Studtio"

**Comment by hunterwerlla** [comment]
When I run this as the first command, it fails to load the command. `SAM Detect` works on my machine as the first command, so I'm not sure if it's an issue in how the command is registered or an issue with my machine, can you validate?

**Comment by bryceitoc9** [comment]
Could you also stamp the info to the top of the toolkit's log file when the extension is activated? Do it right after the `activateLogger()` call in `extension.ts`

File: src/shared/extensionUtilities.ts [hunk 5]
```diff
@@ -202,3 +203,33 @@ export function toastNewUser(context: vscode.ExtensionContext): void {
...
 }
+
+/**
+ * Creates a modal to display OS, AWS Toolkit, and VS Code
+ * versions and allows user to copy to clipboard
+ * Also prints to the toolkit output channel
+ *
+ * @param toolkitOutputChannel VS Code Output Channel
+ */
+
```
Comment (justinmk3) [inline src/shared/extensionUtilities.ts]: unnecessary empty line

**Comment by gsweetwood** [comment]
@hunterwerlla I've added the command to the activation events in the latest commit.  The command should now work, even if you have not connected to AWS yet

File: src/shared/extensionUtilities.ts (hunk 5 above)
Comment (gsweetwood) [reply src/shared/extensionUtilities.ts]: Thanks, I've taken that out in the last commit.

File: package.nls.json (hunk 1 above)
Comment (gsweetwood) [reply package.nls.json]: Changed to "About AWS Toolkit"

File: .changes/next-release/Feature-546b77ba-20be-4573-811d-6a6d86b47f00.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+    "type": "Feature",
+    "description": "Added 'aws.aboutToolkit' command to show AWS Toolkit versioning details that are useful to include with bug reports."
```
Comment (awschristou) [inline .changes/next-release/Feature-546b77ba-20be-4573-811d-6a6d86b47f00.json]: Oops, I meant the user facing command text.  Users don't know what "aws.aboutToolkit" is or how to access it.

File: src/extension.ts [hunk 7]
```diff
@@ -56,6 +61,8 @@ export async function activate(context: vscode.ExtensionContext) {
 
     ext.context = context
     await activateLogger(context)
+    const toolkitEnvDetails = getToolkitEnvironmentDetails()
```
Comment (awschristou) [inline src/extension.ts]: Move this into the try block to reduce the risk of a bad extension activation

Comment (gsweetwood) [reply src/extension.ts]: Thanks, I moved the call into the try block.

File: .changes/next-release/Feature-546b77ba-20be-4573-811d-6a6d86b47f00.json (hunk 6 above)
Comment (gsweetwood) [reply .changes/next-release/Feature-546b77ba-20be-4573-811d-6a6d86b47f00.json]: That makes sense, I've changed it and mentioned the menu option in the latest commit.

**Comment by kiiadi** [comment]
why do we have  a pop-up and it spat out to the console? seems a bit redundant...

**Comment by gsweetwood** [comment]
@kiiadi True. If we decide to keep only one, I vote for the modal.  Otherwise it could seem like nothing happens if the user does not have the output panel open.

**Pull request merged**